pkg/controller/activegate/activegate_controller.go

```diff
@@ -113,10 +115,17 @@ func (r *ReconcileActiveGate) Reconcile(request reconcile.Request) (reconcile.Re
 		return agerrors.HandleSecretError(secret, err, reqLogger)
 	}
 
-	// Define a new Pod object
-	log.Info("creating new pod definition from custom resource")
+	dtc, err := r.dtcBuildFunc(r.client, instance, secret)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = r.reconcilePullSecret(*instance, reqLogger, dtc)
```

Comment:
Isn't this optional if a custom image pull is not set?

Reply:
Added a check if the image and the custompullsecret are empty